src/server.cpp

```diff
@@ -29,22 +29,53 @@ server::listen(int port, const std::string& address)
         _app.start(_concurrency);
     }
 
-    auto ip_address = asio::ip::make_address(address);
+    std::vector<std::string> resolve_addresses;
+    if (address.empty()) {
+        resolve_addresses = {"::", "0.0.0.0"};
+    } else {
+        resolve_addresses = {address};
+    }
+
+    // Two separate connections for both IPv4 and IPv6 are always created. On some operating
+    // systems depending on how the operating system is configured it's possible to open
+    // a single IPv6 socket and let it listen for IPv4 connections too. However, there is
+    // no way to query whether this behavior is enabled on runtime, so a IPv6-only sockets
+    // are forced instead (see acceptor::acceptor())
+    boost::asio::ip::tcp::resolver resolver{_app.ioc()};
+    for (const auto& resolve_address : resolve_addresses) {
+        boost::asio::ip::tcp::resolver::query query{resolve_address, std::to_string(port)};
+
+        // Resolving synchronously because errors should be propagated directly to
+        // the caller of listen()
+        auto resolved = resolver.resolve(query);
```

Comment:
It seems that this one is marked as deprecated (resolver.resolve).

Reply:
Ugh, I didn't notice this, probably because I used older boost. I'll fix this and get back to you.